src/UrlHelper.php

```diff
@@ -41,16 +48,24 @@ public function __construct(RouterInterface $router)
      * Generate a URL based on a given route.
      *
      * @param string $routeName
-     * @param array $params
+     * @param array  $routeParams
+     * @param array  $queryParams
+     * @param string $fragmentIdentifier
+     * @param array  $options       Can have the following keys:
+     *                              - router (array): contains options to be passed to the router
+     *                              - reuse_result_params (bool): indicates if the current RouteResult
+     *                              parameters will be used, defaults to true
+     *
      * @return string
-     * @throws Exception\RuntimeException if no route provided, and no result match
-     *     present.
-     * @throws Exception\RuntimeException if no route provided, and result match is a
-     *     routing failure.
-     * @throws RouterException if router cannot generate URI for given route.
+     * @throws \Zend\Expressive\Helper\Exception\RuntimeException
      */
-    public function __invoke($routeName = null, array $params = [])
-    {
+    public function __invoke(
+        $routeName = null,
+        array $routeParams = [],
+        array $queryParams = [],
+        $fragmentIdentifier = null,
+        array $options = []
+    ) {
```

Comment:
Oh, nice! I missed somewhere along the way that the _consumption_ of the helper is BC with the 1.0 and 2.0 series! That makes the consumer side much easier!